### PR TITLE
Experiment Table Cleanup

### LINKED ISSUE #526: Consolidate experiment table
### Describe the documentation issue

As we increase the number of workloads through tt-blacksmith, the [main experiments table ](https://docs.tenstorrent.com/tt-blacksmith/src/experiments.html#available-experiments) is becoming more and more difficult to read.

### Describe the documentation solution you want to see

I propose to:
- keep one row per model (and finetuning setup - whether it's LoRA/SFT/...) - we have a great info/tables for different setups inside the readme.
- link to the relevant part of readme, since we have a single readme for a model family and all links go to the beginning of the readme.

### PR DESCRIPTION
**Pull request by tstepanovicTT**
### Issue
Closes https://github.com/tenstorrent/tt-blacksmith/issues/526

### Problem description
The experiment table is becoming increasingly hard to read due to many experiments logged. 

### What's Changed

- Dataset and device info moved to the corresponding README files. Now the table contains frameworks, model names, fine tuning methods, and links to READMEs.

- One row kept per model + method (in a given framework). There were multiple lines for experiments using the same model and method but different parallelism strategies. Strategy info (input_sharding_dim) moved to README configuration tables, together with info from the previous point.

### Checklist
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: docs/src/experiments.md [hunk 1]
```diff
@@ -4,71 +4,53 @@ This page provides an overview of the experiments included in this repository, d
...
-- `experiments/`: Experiments are organized first by the framework they utilize, and then by the specific model or task. For example, the PyTorch-based MNIST experiment can be found under `experiments/torch/mnist/`. Within each experiment directory, you will typically find the following files:
+- `experiments/`: Experiments are organized first by the framework they utilize, and then by the specific model or task. For example, the PyTorch-based MNIST experiment can be found under `experiments/torch/mnist/`. Within each experiment directory, you will typically find the following subdirectories and files:
 
+    - Subdirectories named after fine-tuning methods used in experiments (e.g. `lora`, `dpo`, `adapters`). 
+    - Subdirectories specifying the compute environment (e.g. `single_chip`, `quietbox`, `loudbox`, `galaxy`) or sharding strategy (`data_parallel`, `tensor_parallel`). If sharding strategy isn't specified, the single chip configuration is assumed.
```
Comment (mihailomilosevic2001) [inline docs/src/experiments.md]: We should not suggest leaving `data_parallel` and `tensor_parallel` directories. Could you just leave per hardware, then we might in future fix up experiments that have different structure

File: blacksmith/experiments/torch/mnist/README.md [hunk 2]
```diff
@@ -39,6 +39,12 @@ python blacksmith/experiments/torch/mnist/tensor_parallel/train.py
 
 ## Configuration
 
+| Architecture       | mesh_shape                   | mesh_axis_names      | input_sharding_dim | dataset      | Method                      |
+| ------------------ | ---------------------------- | -------------------- | ------------------ | ------------ | --------------------------- |
+| [Single-Chip](single_chip/mnist.yaml) | None      | None                 | None               | MNIST        | Full Model                  |
+| [N300](data_parallel/mnist_dp.yaml)   | `[1, 2]`  | `["model", "batch"]` | `["batch"]`        | MNIST        | Full Model, Data Parallel   |
```
Comment (mihailomilosevic2001) [inline blacksmith/experiments/torch/mnist/README.md]: input_sharding_dim is a string, not a list

File: blacksmith/experiments/torch/llama/xla/lora/README.md [hunk 3]
```diff
@@ -139,9 +139,9 @@ python3 blacksmith/experiments/torch/llama/xla/train.py --config blacksmith/expe
 
 #### Llama 3.3 8B Instruct Training Configurations
 
-| Architecture       | mesh_shape | mesh_axis_names                          | dataset | Method |
-| ------------------ | ---------- | ---------------------------------------- | ------- | ------ |
-| [Wormhole QuietBox](quietbox/llama_3_1_8b_instruct_metamathqa.yaml) | `[2, 4]`   | `["data", "model"]`                      | MetaMathQA    | LoRA   |
+| Architecture       | mesh_shape                   | mesh_axis_names      | input_sharding_dim | dataset      | Method                      |
+| ------------------ | ---------------------------- | -------------------- | ------------------ | ------------ | --------------------------- |
+| [Wormhole QuietBox](quietbox/llama_3_1_8b_instruct_metamathqa.yaml) | `[2, 4]`   | `["data", "model"]`  | `["data"]`  | MetaMathQA    | LoRA   |
```
Comment (mihailomilosevic2001) [inline blacksmith/experiments/torch/llama/xla/lora/README.md]: ditto, not a list

File: blacksmith/experiments/jax/mnist/README.md [hunk 4]
```diff
@@ -3,6 +3,12 @@
 This directory contains the code in JAX (and in Flax) for training multilayer perceptron (MLP) on MNIST dataset, through Tenstorrent's Forge compiler with logging to [Weights&Biases](https://wandb.ai/site/).
 The connection to the Tenstorrent device is established in script ```blacksmith/tools/jax_utils.py```. For more details on connecting to Tenstorrent's hardware in JAX, please refer to [tt-xla](https://github.com/tenstorrent/tt-xla).
 ## Training
+| Architecture       | mesh_shape                   | mesh_axis_names      | dataset      | Method     |
+| ------------------ | ---------------------------- | -------------------- | ------------ | ---------- |
+| [Single-Chip](mnist.yaml) | None               | None      | MNIST         | Full Model       |
+| [N300](mnist.yaml) | `[2]`                     | `["dp"]`  | MNIST        | Full Model, Data Parallel       |
```
Comment (mihailomilosevic2001) [inline blacksmith/experiments/jax/mnist/README.md]: @umalesTT @pglusacTT  what do you guys think about this? We saw that easyDel uses single dimension like this. Jax literally returns the list of devices and these mnist experiments did not really do something special to separate to 2 dimensions.

**Review by mihailomilosevic2001 — COMMENTED** [review]
Good work, left some comments to address 😄

**Comment by mihailomilosevic2001** [comment]
Please in PR description write: Closes link-to-issue
Attaching link to text might not close issue on merge.

**Comment by mihailomilosevic2001** [comment]
Also, please run pre-commit on the PR to pass the tests.

**Comment by mmilosevicTT** [comment]
For what's changed section, I prefer adding bulletpoints for changes added.

**Comment by mmilosevicTT** [comment]
Use title case for PR title and refactor to main.

File: blacksmith/experiments/jax/mnist/README.md (hunk 4 above)
Comment (umalesTT) [reply blacksmith/experiments/jax/mnist/README.md]: Sounds good

File: blacksmith/experiments/torch/llama/xla/lora/README.md [hunk 5]
```diff
@@ -66,14 +66,14 @@ python3 blacksmith/experiments/torch/llama/xla/train.py --config blacksmith/expe
...
-| [Wormhole QuietBox](quietbox/llama_3_2_1b_sst2.yaml) | `[1, 8]`, `[8, 1]`, `[2, 4]` | `["data", "model"]`, `["model", "data"]` | SST2    | LoRA   |
-| [Blackhole QuietBox](quietbox/llama_3_2_1b_sst2.yaml) | `[1, 4]`                     | `["data", "model"]`                      | SST2    | LoRA   |
-| [Galaxy](galaxy/llama_3_2_1b_sst2.yaml) | `[8, 4]`                     | `["data", "model"]`, `["model", "data"]` | SST2    | LoRA   |
+| Architecture       | mesh_shape                   | mesh_axis_names      | input_sharding_dim | dataset      | Method                      |
+| ------------------ | ---------------------------- | -------------------- | ------------------ | ------------ | --------------------------- |
+| [Single-Chip](single_chip/llama_3_2_1b_alpaca.yaml) | None          | None            | None                     | Alpaca  | LoRA   |
+| [Single-Chip](single_chip/llama_3_2_1b_sst2.yaml) | None                         | None        | None                      | SST2    | LoRA   |
+| [N300](quietbox/llama_3_2_1b_sst2.yaml) | `[1, 2]`, `[2, 1]`           | `["data", "model"]`, `["model", "data"]` | `"batch"`  | SST2    | LoRA   |
```
Comment (mmilosevicTT) [inline blacksmith/experiments/torch/llama/xla/lora/README.md]:
```suggestion
| [N300](quietbox/llama_3_2_1b_sst2.yaml) | `[1, 2]`, `[2, 1]`           | `["data", "model"]`, `["model", "data"]` | `"data"`  | SST2    | LoRA   |
```

Was this a mistake?

File: blacksmith/experiments/torch/llama/xla/lora/README.md [hunk 6]
```diff
@@ -121,14 +121,14 @@ python3 blacksmith/experiments/torch/llama/xla/train.py --config blacksmith/expe
...
-| [Wormhole QuietBox](quietbox/llama_3_1_8b_sst2.yaml) | `[8, 1]`   | `["model", "data"]`                      | SST2    | LoRA   |
-| [Wormhole QuietBox](quietbox/llama_3_1_8b_sst2.yaml) | `[2, 4]`   | `["data", "model"]`                      | SST2    | LoRA   |
-| [Blackhole QuietBox](quietbox/llama_3_1_8b_sst2.yaml) | `[1, 4]`   | `["data", "model"]`                      | SST2    | LoRA   |
-| [Galaxy](galaxy/llama_3_1_8b_sst2.yaml) | `[8, 4]`   | `["data", "model"]`, `["model", "data"]` | SST2    | LoRA   |
+| Architecture       | mesh_shape                   | mesh_axis_names      | input_sharding_dim | dataset      | Method                      |
+| ------------------ | ---------------------------- | -------------------- | ------------------ | ------------ | --------------------------- |
+| [P150](single_chip/llama_3_1_8b_sst2.yaml)           | None     | None                            | None            | SST2    | LoRA   |
+| [Wormhole QuietBox](quietbox/llama_3_1_8b_sst2.yaml) | `[1, 8]`   | `["data", "model"]`           | `["batch"]`     | SST2    | LoRA   |
```
Comment (mmilosevicTT) [inline blacksmith/experiments/torch/llama/xla/lora/README.md]:
```suggestion
| [Wormhole QuietBox](quietbox/llama_3_1_8b_sst2.yaml) | `[1, 8]`   | `["data", "model"]`           | `"data"`     | SST2    | LoRA   |
```

Not a list right?

**Review by mmilosevicTT — APPROVED** [review]
I left few more comments, but the PR looks ready to merge 😄

File: blacksmith/experiments/torch/llama/xla/lora/README.md (hunk 5 above)
Comment (tstepanovicTT) [reply blacksmith/experiments/torch/llama/xla/lora/README.md]: input_sharding_dim in the corresponding .yaml file is "batch" but axis names are "model" and "batch"

Comment (tstepanovicTT) [reply blacksmith/experiments/torch/llama/xla/lora/README.md]: will update everything to match the .yaml files

**Review by mmilosevicTT — APPROVED** [review]

**Comment by mmilosevicTT** [comment]
@tstepanovicTT Congrats on your first PR merge 🎉
